python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -1296,9 +1324,23 @@ def _normalize_aggs(
                 columns = values._columns
                 aggs_per_column = (aggs,) * len(columns)
         elif not aggs and kwargs:
-            column_names, aggs_per_column = kwargs.keys(), kwargs.values()
-            columns = tuple(self.obj._data[x[0]] for x in kwargs.values())
-            aggs_per_column = tuple(x[1] for x in kwargs.values())
+            column_names = kwargs.keys()
+
+            def _raise_invalid_type(x):
+                raise TypeError(
+                    f"Invalid keyword argument {x} of type {type(x)} was passed to agg"
+                )
+
+            columns, aggs_per_column = zip(
+                *(
+                    (self.obj._data[x[0]], x[1])
+                    if isinstance(x, tuple)
+                    else (self.obj._data[x.column], x.pyfunc)
+                    if isinstance(x, NamedAgg)
```

Comment:
I don't think we're hitting this condition since an instantiated `NamedAgg` is a tuple instance

```python
In [1]: import pandas as pd

In [2]: agg_a = pd.NamedAgg(column="a", aggfunc="min")

In [3]: isinstance(agg_a, tuple)
Out[3]: True
```

Reply:
Thanks for the catch, I deleted the logic